src/App.jsx

```diff
@@ -1,15 +1,60 @@
+import { useState } from 'react';
 import './App.css';
+import ChatLog from './components/ChatLog';
+import messages from './data/messages.json';
 
-const App = () => {
+
+const calculateTotalLikeCount = (chat) => {
+  return chat.reduce((total, chat) => {
+    return total + (chat.liked ? 1 : 0);
+  }, 0);
+};
+
+const localSender = messages[0].sender;
+let remoteSender = 'Unknown';
+
+for (const message of messages) {
+  if (message.sender !== localSender) {
+    remoteSender = message.sender;
+    break; // Stop once other sender is found
+  }
+}
+
+
+const App =() => {
+  const [chat, setChat] = useState(messages);
```

Comment:
Maybe we could change this from `chat` to `chats` since it is multiple chat messages instead of one?